### PR TITLE
access module context from state class (#315)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #315 

This would allow a function to be declared on the main module and that function can be used accross multiple modules
```
def reset_data(state):
    if "page1" in state._module:
        state.df = df
        state.filter_value = 0
        notify(state, message="Table data is reset")
    if "page2" in state._module:
        state.chart_data = df
        state.fv = 0
        notify(state, message="Chart data is reset")
```

### CODE REVIEW

File: taipy/gui/gui_actions.py [hunk 1]
```diff
@@ -134,7 +134,16 @@ def get_context_id(state: State) -> t.Optional[str]:
         return state._gui._get_client_id()
     return None
 
-def invoke_state_callback(gui: Gui, context_id: str, user_callback: t.Callable, args: t.Union[t.Tuple, t.List]) -> t.Any:
+
+def get_page_scope(state: State) -> t.Optional[str]:
+    if state and isinstance(state._gui, Gui):
+        return state._gui._get_locals_context()
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui_actions.py:160]: is this a str ?

**Review by FredLL-Avaiga — COMMENTED** [review]
I'm still not sure I get it ...
gui._get_locals_context() returns the "module" for the UI modified variable ...
is there a module for on_action ?
What happens when the change comes from the back ?

File: taipy/gui/gui_actions.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui_actions.py:160]: Yes it is a string

**Review by FredLL-Avaiga — APPROVED** [review]

File: taipy/gui/gui_actions.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/gui/gui_actions.py:160]: Doc:
```
"""Get the module name that trigged a callback.

Pages can be defined in different modules yet share callbacks declared elsewhere (typically, the
application's main module).

This function returns the name of the module where the page (that holds the control that triggered
the callback) was declared.  This lets applications implement different behaviors depending on what page
is involved.

This function must be called only in the body of a callback function.

Arguments:

    state: (State^): The `State` instance, which is an argument of the callback function.

Returns:

    str: The name of the module that holds the definition of the page containing the control that triggered
          this callback.
"""
```

**Review by FabienLelaquais — APPROVED** [review]
👍

File: taipy/gui/gui_actions.py [hunk 2]
```diff
@@ -134,7 +134,16 @@ def get_context_id(state: State) -> t.Optional[str]:
         return state._gui._get_client_id()
     return None
 
-def invoke_state_callback(gui: Gui, context_id: str, user_callback: t.Callable, args: t.Union[t.Tuple, t.List]) -> t.Any:
+
+def get_module_name_from_state(state: State) -> t.Optional[str]:
```
Comment (FabienLelaquais) [inline taipy/gui/gui_actions.py:138]: Doc:
```
"""Get the module name that trigged a callback.

    Pages can be defined in different modules yet share callbacks declared elsewhere (typically, the
    application's main module).

    This function returns the name of the module where the page (that holds the control that triggered the
    callback) was declared.  This lets applications implement different behaviors depending on what page
    is involved.

    This function must be called only in the body of a callback function.

    Arguments:

        state: (State^): The `State` instance, which is an argument of the callback function.

    Returns:

        str: The name of the module that holds the definition of the page containing the control that
            triggered this callback.
"""
```

**Pull request merged**